src/node_process_methods.cc

```diff
@@ -471,6 +473,128 @@ static void ReallyExit(const FunctionCallbackInfo<Value>& args) {
   env->Exit(code);
 }
 
+inline char** copy_js_strings_array(Environment* env,
+                                    Local<Array> js_array,
+                                    int* target_length) {
+  Local<Context> context = env->context();
+  char** target = nullptr;
+  int length = js_array->Length();
+
+  CHECK_LT(length, INT_MAX);
+
+  target = new char*[length + 1];
+  target[length] = nullptr;
+
+  if (length > 0) {
+    for (int i = 0; i < length; i++) {
+      node::Utf8Value str(env->isolate(),
+                          js_array->Get(context, i).ToLocalChecked());
+      target[i] = strdup(*str);
+      CHECK_NOT_NULL(target[i]);
+    }
+  }
+
+  if (target_length != nullptr) {
+    *target_length = length;
+  }
+
+  return target;
+}
+
+#if !defined(_WIN32)
+inline int persist_standard_stream(int fd) {
+  int flags = fcntl(fd, F_GETFD, 0);
+
+  if (flags < 0) {
+    return flags;
+  }
+
+  flags &= ~FD_CLOEXEC;
+  return fcntl(fd, F_SETFD, flags);
+}
+
+static void Replace(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Local<Context> context = env->context();
+
+  CHECK_GT(args.Length(), 1);
+  CHECK(args[0]->IsString());
+
+  // Prepare arguments and environment
+  char** argv = nullptr;
+  char** envp = nullptr;
+  int envp_length = 0;
+
+  // Prepare arguments - Note that the pathname is always the first argument
+  if (args.Length() > 1) {
+    CHECK(args[1]->IsArray());
+
+    Local<Array> argv_array = args[1].As<Array>();
+
+    Local<Array> full_argv_array =
+        Array::New(env->isolate(), argv_array->Length() + 1);
+    full_argv_array->Set(context, 0, args[0].As<String>()).Check();
+    for (unsigned int i = 0; i < argv_array->Length(); i++) {
+      full_argv_array
+          ->Set(context, i + 1, argv_array->Get(context, i).ToLocalChecked())
+          .Check();
+    }
```

Comment:
Does this make a copy? I didn't quite understand the goal here. Sorry for the questions!

Reply:
Yes, that's the idea. `execve` wants a null-terminated `char**` while I have a `JsArray`.
On the Javascript side I have validated they are all strings with no null-bytes in the middle.
Any better idea on how to handle this?

Reply:
My gut feeling would be that it's much easier to handle the conditional and array copying on the JS side of things and instead only pass a version of the array(s) down to C++ that's close to what the syscall expects 🙂

Reply:
I see what you mean now.
I would love to make all the manipulation on the JS side, but it is then possible to pass the data from JS to be interpreted on C++ as `char**`. Out of my mind I'm thinking about using `Uint8Array` but I might be wrong.

Reply:
You can still pass an array of strings, it's just the array copying/argument handling that would end up simplified (i.e. no need to check `if (args.Length() > 1) {`, no need to `full_argv_array->Set(context, i + 1, argv_array->Get(context, i).ToLocalChecked())`)

Reply:
This function has been removed.